File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length, dp[] = new int[n+2];
        Arrays.fill(dp, Integer.MAX_VALUE);
        dp[0] = dp[1] = 0;
        for(int i = 0; i < n; i++) {
            dp[i+1] = Math.min(dp[i+1], dp[i] + cost[i]);
            dp[i+2] = Math.min(dp[i+2], dp[i] + cost[i]);
        }
        return Math.min(dp[n], dp[n+1]);
    }
}

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        dp = [inf] * (n+2)
        dp[0] = dp[1]=0
        for i in range(n):
            dp[i+1] = min(dp[i+1], dp[i]+cost[i])
            dp[i+2] = min(dp[i+2], dp[i]+cost[i])
        return min(dp[n], dp[n+1])